### ISSUE TITLE
rsyslog stops writing logs until restarted

### ISSUE THREAD

**Issue by Uni-lovenix** [rsyslog/rsyslog#3860]
rsyslog: 4.6.2
Redhat x86_64

We have 4 clients that send logs to 1 remote hosts via tcp. But only one client sends log to remote host. and the only one sends log it's very small. I don't know why others client stop sending logs to remote host suddenly.  
PS: 
For some other reason, we are temporarily unable to upgrade this tool.

### rsyslog configuration

$FailOnChownFailure off

$ModLoad imuxsock.so	# provides support for local system logging (e.g. via logger command)
$ModLoad imklog.so	# provides kernel logging support (previously done by rklogd)

$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat

kern.warning;*.err;authpriv.none        		/dev/tty10

*.info;mail.none;authpriv.none;cron.none                /var/log/messages

authpriv.*                                              /var/log/secure

mail.*                                                  -/var/log/maillog

cron.*                                                  /var/log/cron

uucp,news.crit                                          /var/log/spooler

local7.*                                                /var/log/boot.log

$ModLoad imfile

$InputFileName /usr/local/app/stream/logs/gunicorn_error.log
$InputFileTag tag1:stream
$InputFileStateFile /usr/local/app/stat-file1

$InputFileSeverity info
$InputFileFacility local7
$InputRunFileMonitor

$InputFilePollInterval 1
$InputFilePersistStateInterval 1

*.* @@test.com:8891

**Comment by davidelang** [comment]
is this really rsyslog v4? that's over a decade old. Please try a current version (we fixed similar bugs in v8.1908)

**Comment by Uni-lovenix** [comment]
> is this really rsyslog v4? that's over a decade old. Please try a current version (we fixed similar bugs in v8.1908)

Yeah, For some other reason, we are temporarily unable to upgrade this tool.
